understanding/sharding.md

```diff
@@ -2,29 +2,210 @@
+
+### Example Sharding Manager Code
 ```javascript
 /*
     The following code goes into it's own file, and you run this file
     instead of your main bot file.
 */
 
-const Discord = require('discord.js');
-const Manager = new Discord.ShardingManager('./YOUR_BOT_FILE_NAME.js');
-Manager.spawn(2); // This example will spawn 2 shards (5,000 guilds);
+// Include discord.js ShardingManger
+const { ShardingManager } = require('discord.js');
+
+// Create your ShardingManger instance
+const manager = new ShardingManager('./YOUR_BOT_FILE_NAME.js', {
+    // for ShardingManager options see:
+    // https://discord.js.org/#/docs/main/v12/class/ShardingManager
+    totalShards: 'auto',
+    token: 'YOUR_TOKEN_GOES_HERE'
+});
+
+// Spawn your shards
+manager.spawn();
+
+// Emitted when a shard is created
+manager.on('shardCreate', (shard) => console.log(`Shard ${shard.id} launched`));
 ```
 
+## Sharing Information Between Shards
+
+Remember how we were talking about sharding being a method of "splitting" the bot into multiple instances of itself? Because of this, things like your adding your total guilds or getting a specific guild are not as simple as they were before. We must now use either [`fetchClientValues`](#fetchclientvalues) or [`broadcastEval`](#broadcasteval) to get information from across shards.
+
+These two functions are your go-to for getting any information from other shards, so get familiar with them!
+
+### FetchClientValues
+
+[`fetchClientValues`](https://discord.js.org/#/docs/main/v12/class/ShardClientUtil?scrollTo=fetchClientValues) gets Client properties from all shards. This is what you should use when you would like to get any of the nested properties of the Client, such as `guilds.size` or `uptime`. It's useful for getting things like Collection sizes, basic client properties, and unprocessed information about the client.
+
+Example:
+```javascript
+/*
+    Example result of fetchClientValues() on a bot with 4,300 guilds split across 4 shards.
+    Assume this is being executed on shard 0, the first shard.
+*/
+
+// If we just get our client.guilds.cache.size, it will return
+// only the number of guilds on the shard this is being run on.
+console.log(client.guilds.cache.size);
+// 1050
+
+// If we would like to get our client.guilds.cache.size from all
+// of our shards, we must make use of fetchClientValues().
+const res = await client.shard.fetchClientValues('guilds.cache.size');
+console.log(res);
+// 	[
+//		1050,	// shard 0
+//		1100,	// shard 1
+//		1075,	// shard 2
+//		1075	// shard 3
+//	]
+
+````
+
+Let's say you want to do something like get your total server count - In a non-sharded environment, this would be as simple as getting the `client.guilds.cache.size`. However, in this case `client.guilds.cache.size` will not return the total servers your bot is in. Instead it returns only the total number of servers *on this shard*, like in the first part of the example above.
+
```

Comment:
I feel it's a bit redundant to have both the comments, and this paragraph, showing the same thing, and then below to show a real-life example of using the code above. This can be made more concise by removing the comments and just having the text.